service/frontend/service.go

```diff
@@ -200,6 +200,8 @@ type Config struct {
 	CallbackURLMaxLength        dynamicconfig.IntPropertyFnWithNamespaceFilter
 	MaxCallbacksPerWorkflow     dynamicconfig.IntPropertyFnWithNamespaceFilter
 	AdminEnableListHistoryTasks dynamicconfig.BoolPropertyFn
+
+	MaskInternalOrUnknownErrors dynamicconfig.BoolPropertyFn
```

Comment:
Plz follow existing pattern of initializing this field in `NewConfig` below and use `serviceConfig *Config` in `MaskInternalErrorsInterceptorProvider`

Reply:
or maybe just remove this field here.

I am actually not sure about the recommended way after the dc refactor. cc @dnr what's your suggestion here?

Reply:
I don't have a strong preference. Personally I find the giant struct kind of silly, but I also don't want to change it now. So yeah, let's keep this one in serviceConfig too.

But more importantly: the interceptor constructor should accept a `dynamicconfig.BoolPropertyFnWithNamespaceFilter`, not a whole collection. That'll make it easier to test, and also possible to reuse the interceptor for other services. Only service/frontend should reference the Setting, the interceptor shouldn't touch dynamic config at all, just use the function it's given.

Reply:
change constructor to accept a function.